static-assets/components/cstudio-forms/template-editor.js

```diff
@@ -395,7 +395,7 @@ CStudioAuthoring.Module.requireModule(
 
                         aceEditor.getSession().on('change', function() {
                           aceEditor.isModified = true;
-                          onSaveCb.pendingChanges(true);
+                          onSaveCb.pendingChanges?.(true);
```

Comment:
When opening the editor from the content-type, it doesn't use the embedded_legacy_form, so it doesn't need/use pendingChanges. That was causing the strange behavior